AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOrchestrator.swift

```diff
@@ -150,6 +154,10 @@ final class AWSInitialSyncOrchestrator: InitialSyncOrchestrator {
     }
 
     private func makeCompletionResult() -> Result<Void, DataStoreError> {
+        if syncErrors.allSatisfy(isUnauthorizedError) {
+            return .successfulVoid
+        }
```

Comment:
what happens if syncErrors is empty, does it return true for `syncErrors.allSatisfy(isUnauthorizedError`? seems like logically this can also be inside the guard statement like
```
guard syncErrors.isEmpty else {
  if syncErrors.allSatisfy(isUnauthorizedError) {
            return .successfulVoid
     } else {
    // return failure
   }
} 
```

Reply:
`allSatisfy` returns true for an empty array, refactored to return early